platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -82,29 +75,64 @@ private static int parseSystemProperty(String propertyKey, int defaultValue) {
     }
 
+    /**
+     * Creates a new {@link CloudSolrClient} using the list of Zookeeper hosts and collection name
+     * provided.
+     *
+     * @param zookeeperHosts comma-separate list of Zookeeper hosts managing the Solr Cloud
+     *                       configuration
+     * @param collection     name of the collection to create
+     * @return {@code Future} used to retrieve the new {@link CloudSolrClient} instance
+     */
     public static Future<SolrClient> getClient(String zookeeperHosts, String collection) {
-        RetryPolicy retryPolicy = new RetryPolicy()
-                .retryWhen(null)
-                .withBackoff(10,
-                TimeUnit.MINUTES.toMillis(1),
-                TimeUnit.MILLISECONDS);
+        RetryPolicy retryPolicy = new RetryPolicy().retryWhen(null)
+                .withBackoff(10, TimeUnit.MINUTES.toMillis(1), TimeUnit.MILLISECONDS);
         return Failsafe.with(retryPolicy)
                 .with(EXECUTOR_SERVICE)
                 .onRetry((c, failure, ctx) -> LOGGER.debug(
-                        "Attempt {} failed to create Solr Cloud client ({}). Retrying again.",
+                        "Attempt {} failed to create Solr Cloud client for collection {} using Zookeeper hosts [{}]. Retrying again.",
                         ctx.getExecutions(),
-                        collection))
+                        collection,
+                        zookeeperHosts))
                 .onFailedAttempt(failure -> LOGGER.debug(
-                        "Attempt failed to create Solr Cloud client (" + collection + ")",
+                        "Attempt failed to create Solr Cloud client for collection {} using Zookeeper hosts [{}]",
+                        collection,
+                        zookeeperHosts,
                         failure))
-                .onSuccess(client -> LOGGER.debug("Successfully created Solr Cloud client ({})",
+                .onSuccess(client -> LOGGER.debug(
+                        "Successfully created Solr Cloud client for collection {}",
                         collection))
                 .onFailure(failure -> LOGGER.warn(
-                        "All attempts failed to create Solr Cloud client (" + collection + ")",
+                        "All attempts failed to create Solr Cloud client for collection {} using Zookeeper host [{}]",
+                        collection,
+                        zookeeperHosts,
                         failure))
                 .get(() -> createSolrCloudClient(zookeeperHosts, collection));
     }
 
+    private static ScheduledExecutorService createExecutorService() throws NumberFormatException {
+        Integer threadPoolSize = parseSystemProperty("org.codice.ddf.system.threadPoolSize",
+                THREAD_POOL_DEFAULT_SIZE);
+        return Executors.newScheduledThreadPool(threadPoolSize);
+    }
+
+    private static int parseSystemProperty(String propertyKey, int defaultValue) {
```

Comment:
I think we could replace this function implementation with `NumberUtils.toInt(System.getProperty(propertyKey), defaultValue)`.

https://commons.apache.org/proper/commons-lang/javadocs/api-3.5/org/apache/commons/lang3/math/NumberUtils.html#toInt-java.lang.String-int-